infra/modules/azure_core_infra/_modules/nat_gateway/nat_gateway.tf

```diff
@@ -0,0 +1,31 @@
+resource "azurerm_public_ip_prefix" "ng" {
+  count = var.ng_ippres_number
+
+  name                = format("%s-ippre-%02d", "${var.project}-ng", count.index + 1)
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  prefix_length = 31
+  zones         = [count.index + 1]
+
+  tags = var.tags
+}
+
+resource "azurerm_nat_gateway" "this" {
+  count = var.ng_number
```

Comment:
Does this have sense? Because we already let client set the variable `var.nat_enabled`. Am I missing any use case?

Reply:
The `ng_number` parameter is not used to enable or disable features; instead, it determines the number of instances.
Here the variable with validation:
```hcl
variable "ng_number" {
  type        = number
  description = "Number of nat gateways to deploy"
  default     = 1

  validation {
    condition     = var.ng_number > 0 && var.ng_number <= 9
    error_message = "Number of nat gateways must be between 1 and 9."
  }
}
```